extensions/network/configurationinternal.m

```diff
@@ -328,6 +328,38 @@ static int dynamicStoreLocation(lua_State *L) {
     return 1 ;
 }
 
+/// hs.network.configuration:locations() -> table
+/// Method
+/// Returns all configured locations
+///
+/// Parameters:
+///  * None
+///
+/// Returns:
+///  * a table of key-value pairs mapping location UUIDs to their names
+///
+static int dynamicStoreLocations(lua_State *L) {
+    LuaSkin *skin = [LuaSkin shared] ;
+    [skin checkArgs:LS_TUSERDATA, USERDATA_TAG, LS_TBREAK] ;
+    SCPreferencesRef prefs = SCPreferencesCreate(NULL, CFSTR("Hammerspoon"), NULL);
+
+    CFArrayRef locations = SCNetworkSetCopyAll(prefs);
```

Comment:
likewise with this: `if (locations) { ... }` and move the `CFRelease(locations)` inside the then block.